src/homework_8/myArrayList/MyArrayList.java

```diff
@@ -0,0 +1,138 @@
+package homework_8.myArrayList;
+
+import java.util.Arrays;
+
+public class MyArrayList<T> implements List<T> {
+    private static final Object[] EMPTY_ARRAY = {};
+    private static final Object[] FULL_ARRAY_CAPACITY = new Object[10];
+    private T[] array;
+
+    public MyArrayList() {
+        array = (T[]) FULL_ARRAY_CAPACITY;
+    }
+
+    public MyArrayList(int capacity) {
+        if (capacity > 0) {
+            array = (T[]) new Object[capacity];
+        } else if (capacity == 0) {
+            array = (T[]) EMPTY_ARRAY;
```

Comment:
better write:
private static final  int EMPTY_ARRAY = 0;  // in the class
array = (T[]) new Objects[EMPTY_ARRAY];  // in constructor
and remove:
private static final Object[] EMPTY_ARRAY = {};